tests/integration/injection-test.js

```diff
@@ -1,118 +1,71 @@
-import {
-  setup as setupModelFactoryInjections,
-  reset as resetModelFactoryInjections,
-} from 'dummy/tests/helpers/model-factory-injection';
-import EmberObject from '@ember/object';
-import { getOwner } from '@ember/application';
-import { run } from '@ember/runloop';
-import setupStore from 'dummy/tests/helpers/store';
-import DS from 'ember-data';
+import Model from 'ember-data/model';
+import Service from '@ember/service';
 import { module, test } from 'qunit';
-
-let env, hasFactoryFor, originalLookupFactory, originalOwnerLookupFactory, originalFactoryFor;
-
-const model = {
-  isModel: true,
-  _create() {},
-};
-const factory = {
-  class: model,
-};
-
-module('integration/injection factoryFor enabled', {
-  beforeEach() {
-    env = setupStore();
-
-    if (getOwner) {
-      let owner = getOwner(env.store);
-
-      hasFactoryFor = !!owner.factoryFor;
-      originalFactoryFor = owner.factoryFor;
-      originalOwnerLookupFactory = owner._lookupFactory;
-
-      if (originalFactoryFor) {
-        owner.factoryFor = function(/* name */) {
-          return factory;
-        };
-      } else {
-        // Ember 2.3 - Ember 2.11
-        originalOwnerLookupFactory = owner._lookupFactory;
-        owner._lookupFactory = function() {
-          return model;
-        };
-      }
-    } else {
-      originalLookupFactory = env.store.container.lookupFactory;
-      env.store.container.lookupFactory = function() {
-        return model;
-      };
-    }
-  },
-
-  afterEach() {
-    if (getOwner) {
-      let owner = getOwner(env.store);
-
-      if (owner.factoryFor) {
-        owner.factoryFor = originalFactoryFor;
-      } else {
-        owner._lookupFactory = originalOwnerLookupFactory;
-      }
-    } else {
-      env.store.container.lookupFactory = originalLookupFactory;
-    }
-
-    run(env.store, 'destroy');
-  },
+import { setupTest } from 'ember-qunit';
+
+module('integration/injection factoryFor enabled', function(hooks) {
+  setupTest(hooks);
+  let store;
+  let Model;
+  let factory;
+
+  hooks.beforeEach(function() {
+    let { owner } = this;
+    Model = {
+      isModel: true,
+    };
+    owner.register('model:super-villain', Model);
+    store = owner.lookup('service:store');
+
+    let originalFactoryFor = owner.factoryFor;
+
+    owner.factoryFor = function interceptFactoryFor() {
```

Comment:
This seems really bad... Why do we need to do this?

Reply:
Unless I'm missing something, we can remove this whole thing (and the `factory` variable)...

Reply:
converted to using `owner.factoryFor` to get the factory in the test.